pageserver/src/bin/pageserver.rs

```diff
@@ -303,62 +303,73 @@ fn start_pageserver(conf: &'static PageServerConf) -> anyhow::Result<()> {
             .build()
             .map_err(|err| anyhow!(err))?;
         let service = utils::http::RouterService::new(router).unwrap();
         let server = hyper::Server::from_tcp(http_listener)?
             .serve(service)
-            .with_graceful_shutdown(task_mgr::shutdown_watcher());
+            .with_graceful_shutdown(cancellation_token.cancelled());
 
         task_mgr::spawn(
             MGMT_REQUEST_RUNTIME.handle(),
-            TaskKind::HttpEndpointListener,
-            None,
-            None,
             "http endpoint listener",
             true,
             async {
                 server.await?;
                 Ok(())
             },
         );
+    }
 
-        if let Some(metric_collection_endpoint) = &conf.metric_collection_endpoint {
-            task_mgr::spawn(
-                MGMT_REQUEST_RUNTIME.handle(),
-                TaskKind::MetricsCollection,
-                None,
-                None,
-                "consumption metrics collection",
-                true,
-                async move {
-                    pageserver::consumption_metrics::collect_metrics(
-                        metric_collection_endpoint,
-                        conf.metric_collection_interval,
-                        conf.id,
-                    )
-                    .instrument(info_span!("metrics_collection"))
-                    .await?;
-                    Ok(())
-                },
-            );
-        }
+    if let Some(metric_collection_endpoint) = &conf.metric_collection_endpoint {
+        let metrics_cxt = RequestContext::new(
+            TaskKind::MetricsCollection,
+            DownloadBehavior::Error, // metrics collector shouldn't be downloading anything
+        );
+        task_mgr::spawn(
+            MGMT_REQUEST_RUNTIME.handle(),
+            "consumption metrics collection",
+            true,
+            async move {
+                pageserver::consumption_metrics::collect_metrics(
+                    metric_collection_endpoint,
+                    conf.metric_collection_interval,
+                    conf.id,
+                    metrics_cxt,
+                )
+                .instrument(info_span!("metrics_collection"))
+                .await?;
+                Ok(())
+            },
+        );
     }
 
     // Spawn a task to listen for libpq connections. It will spawn further tasks
     // for each connection. We created the listener earlier already.
-    task_mgr::spawn(
-        COMPUTE_REQUEST_RUNTIME.handle(),
-        TaskKind::LibpqEndpointListener,
-        None,
-        None,
-        "libpq endpoint listener",
-        true,
-        async move {
-            page_service::libpq_listener_main(conf, auth, pageserver_listener, conf.auth_type).await
-        },
-    );
+    {
+        let libpq_cxt = RequestContext::new(
+            TaskKind::LibpqEndpointListener,
+            DownloadBehavior::Error, // listener task shouldn't be downloading anything
```

Comment:
Why shouldn't the pageservice be downloading anything?

Reply:
This context is used only by the listener task that accepts connections. Once accepted, we spawn a new task, and create a new context, for each connection. That context is created with `DownloadBehavior::Download` as you would expect. I'll expand the comment to explain that.

Reply:
Oki, I guess it also binds back to what I was thinking in some other thread that we most likely don't really have disjoint hierarchies, but everything always flows through tenants and timelines. As in, there is no LibpqEndpointListener hierarchy, that is just another aspect of tenant/timeline lifecycle.

The aim seems to be just to use the one infrastructure for the two different things:
- stop accepting during graceful shutdown
- otherwise exist to support detach

I think if these two were separate matters, perhaps in future, it would clarify and simplify. For example, accepting could be stopped just a cancellation token, it does not not need to be related to this request context hierarchies.